cmd/getConfig.go

```diff
@@ -148,11 +149,13 @@ func applyContainerConditions(containers []config.Container, stepConfig *config.
 			for _, param := range container.Conditions[0].Params {
 				if container.Conditions[0].ConditionRef == "strings-equal" && stepConfig.Config[param.Name] == param.Value {
 					var containerConf map[string]interface{}
-					containerConf = stepConfig.Config[param.Value].(map[string]interface{})
-					for key, value := range containerConf {
-						stepConfig.Config[key] = value
+					if stepConfig.Config[param.Value] != nil {
+						containerConf = stepConfig.Config[param.Value].(map[string]interface{})
```

Comment:
This line is not new, but I don't really understand what happens here. Is the value at `param.Value` type-converted to `map[string]interface{}` (and stored in `containerConf`)?

Reply:
correct, basically doing a type assertion:
https://tour.golang.org/methods/15